From 8d1d6c18015de5db2b2a1f499b6d341f86505810 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Sun, 30 Jun 2013 19:51:16 +0200
Subject: [PATCH] url: restore the functionality of 'curl -u :'

This commit fixes a regression introduced in
fddb7b44a79d78e05043e1c97e069308b6b85f79.

Reported by: Markus Moeller
Bug: http://curl.haxx.se/mail/archive-2013-06/0052.html

[upstream commit abca89aaa0fb208cfaf4ead6692014c4e553388a]
---
 lib/url.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/lib/url.c b/lib/url.c
index 7ba4969..4231d37 100644
--- a/lib/url.c
+++ b/lib/url.c
@@ -317,6 +317,13 @@ static CURLcode setstropt_userpwd(char *option, char **userp, char **passwdp,
   if(!result) {
     /* Store the username part of option if required */
     if(userp) {
+      if(!user && option && option[0] == ':') {
+        /* Allocate an empty string instead of returning NULL as user name */
+        user = strdup("");
+        if(!user)
+          result = CURLE_OUT_OF_MEMORY;
+      }
+
       Curl_safefree(*userp);
       *userp = user;
     }
-- 
1.7.1

